students/WooseokJ/MailP2fixedWJFixed.py

```diff
@@ -0,0 +1,121 @@
+
+# coding: utf-8
+
+# In[ ]:
+
+
+import sys
+import datetime
+import operator
+
+
+def menu():
+    print(''''Menu
+    Please choose from below options:
+    1 - Send a Thank You
+    2 - Create a Report
+    3 - Exit the program
+    4 - Print thank you letter to all current donors''')
+
+
+# first it will ask to choose a name whoom you would want to send the mail
+# along side that when typing list it will give you all the names on the current list
+# second if the name is not in the list then it will prompt to add the name into the data base
+# lastly it will print the message of the email
+def send_a_thankyou():
+    donor_list = []
+    for donor in donor_db:
+        donor_list.append(donor.lower())
+    while True:
+        fullname = input("Enter the full name of the donor typing list will give the names of current doners ")
+        if fullname.lower() == 'list':
+            for donor in donor_db:
+                print(donor)
+        elif fullname.lower() not in donor_list:
+            print(f'{fullname} does not exist in the donor data, '
+                  f'we will add {fullname} to the donor data.')
+            amount = float(input("Please enter the donation amount: "))
+            donor_db[fullname] = [amount, 1]
+            print(f"{amount} has been added to {fullname}'s donation history.")
+            break
+        elif fullname.lower() in donor_list:
+            amount = float(input("Please Enter the donation amount: "))
+            donor_db[fullname][0] = donor_db[fullname][0] + amount
+            donor_db[fullname][1] = donor_db[fullname][1] + 1
+            print(f"{amount} has been added to {fullname}'s donation history.")
+            break
+    print("Creating a Thank You email:")
+    print(f'Thank you {fullname} for your donation of {amount:^10.2f}!')
+    print()
+    menu()
+
+
+# this prints out the database of the donors with their total amaounts and average amount donated
+def create_a_report():
+    print("Printing report:")
+    title = ("Donor Name", "| Total Given", "| Num Gifts", "| Average Gift")
+    row = " ".join(["{:<19} | {:^13} | {:^13} | {:^11}"]).format(*title)
+    length = len(row)
+    print("\n" + row)
+    print("=" * length)
+    for key, value in sorted(donor_db.items()):
+        given = str(sum(value))
+        gift = str(len(value))
```

Comment:
Note that this one will always return "2", as that is the number of elements in the _value_ list associated to each donor. Since you're try to get the number of donations, you could do the following:

gift = value[1] # No need to convert to _str_, Python will do that for you when adding the to _row_format_